internal/storage/fs/object/blob/blob_fs.go

```diff
@@ -38,15 +36,38 @@ var _ fs.StatFS = &FS{}
 // ensure FS implements fs.ReadDirFS aka ReadDir
 var _ fs.ReadDirFS = &FS{}
 
+func StrUrl(schema, bucket string) string {
+	return fmt.Sprintf("%s://%s", schema, bucket)
+}
+
 // New creates a FS for the container
-func NewFS(logger *zap.Logger, schema string, containerName string) (*FS, error) {
+func NewFS(ctx context.Context, urlstr string, bucket string, prefix string) (*FS, error) {
+	if prefix != "" {
+		prefix = strings.Trim(prefix, "/") + "/" // to match "a/subfolder/"
+	}
```

Comment:
```suggestion
```
Just a quick though: `strings.Trim` will strip both leading and trailing `"/"`.
I'm wondering if it is better that we don't try and normalize the prefix as a path at all.
That gives the configurer the most control. If we do this they couldn't pick a key with a leading slash, or pick a common prefix that does not end in a trailing slash.
This would also technically be a breaking change, someone might be depending on being able to set a leading slash or omit a trailing one.

Reply:
@GeorgeMac this is the requirement of go-cloud for [prefixed buckets](https://gocloud.dev/howto/blob/#prefix). The library expects the `prefix` to match `a/subfolder/` format.  Their implementation doesn't have any check on prefix and just operates with it. Please check [here](https://github.com/google/go-cloud/blob/97fe9d0425b25c43657f5676799c993e51cd39f0/blob/driver/driver.go#L378-L420)

Reply:
@erka interesting! Thanks for sharing this.

@ahobson does this at all interfere with how you use the s3 backend? I ask as you first implemented the s3 one and wanted to know if you used prefixes with slashes in this way.

Reply:
I'm not on that project any more, but the config just used `a-prefix-name` without slashes, so I'm guessing this would be compatible.

Reply:
@ahobson that sounds like it would not be compatible, no? This change is going to auto append a slash to any prefix provided. So that would become “a-prefix-name/“. Would that cause issues for how it is being used?

Reply:
Just circling back here again: the Go blob store docs are actually inconsistent on their messaging around prefixed buckets. Their examples state it must end in, however, their actual Go doc string suggests it is just a common practice:
https://github.com/google/go-cloud/blob/01064b751a87ca20722835090310a7034a9fc598/blob/blob.go#L1552-L1556
```
PrefixedBucket returns a *Bucket based on b with all keys modified to have prefix, which will usually end with a "/" to target a subdirectory in the bucket.
```

I actually don't think this is a requirement by the blob store themselves:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html
```
There is nothing unique about the slash (/) character, but it is a very common prefix delimiter.
```

So I think we should drop this set of lines and just let the caller decide the prefix.